src/board.rs

```diff
@@ -45,13 +41,123 @@ impl Chessboard {
             black_king: 0b0000100000000000000000000000000000000000000000000000000000000000,
             black_queen: 0b0001000000000000000000000000000000000000000000000000000000000000,
             black_rooks: 0b1000000100000000000000000000000000000000000000000000000000000000,
-            white_castle: 3,
-            black_castle: 3,
+            castling_rights: 0b1111,
+            en_passant: 0,
+            white_turn: true,
+        }
+    }
+
+    /// Create a new instance of a chessboard, with no pieces on it.
+    pub fn empty() -> Chessboard {
+        Chessboard {
+            white_pawns: 0,
+            white_knights: 0,
+            white_bishops: 0,
+            white_king: 0,
+            white_queen: 0,
+            white_rooks: 0,
+            black_pawns: 0,
+            black_knights: 0,
+            black_bishops: 0,
+            black_king: 0,
+            black_queen: 0,
+            black_rooks: 0,
+            castling_rights: 0,
             en_passant: 0,
-            white_turn: true
+            white_turn: true,
         }
     }
 
+    /// Create a new instance of a chessboard, based on a FEN string
+    /// Forsyth–Edwards Notation Parser
+    /// * `fen` - The FEN to be converted to a Chessboard.
+    /// # Return: Chessboard with the position from the FEN.
+    pub fn from_string(fen: &str) -> Chessboard {
+        let mut chessboard = Chessboard::new();
```

Comment:
If we start from a new board (i.e. a board setup for a new game) wouldn't that mess up adding pieces because there are already pieces on the board?